### PR TITLE
Copy satellites in deployment script

### PR DESCRIPTION
**Pull request by cdmihai**
Extracted this from #5222 

When I tried the deployment script msbuild failed not finding the satellite assemblies. Apparently this happens when you're testing msbuild bits with new resources.

### CODE REVIEW

File: scripts/Deploy-MSBuild.ps1 [hunk 1]
```diff
@@ -36,58 +51,65 @@ if ($runtime -eq "Desktop") {
...
-    "artifacts\bin\MSBuild.Bootstrap\$configuration\$targetFramework\Microsoft.VisualBasic.targets"
+    FileToCopy "$bootstrapBinDirectory\Microsoft.Build.dll"
+    FileToCopy "$bootstrapBinDirectory\Microsoft.Build.Framework.dll"
+    FileToCopy "$bootstrapBinDirectory\Microsoft.Build.Tasks.Core.dll"
+    FileToCopy "$bootstrapBinDirectory\Microsoft.Build.Utilities.Core.dll"
+
+    FileToCopy "$bootstrapBinDirectory\en\Microsoft.Build.resources.dll" "en"
+    FileToCopy "$bootstrapBinDirectory\en\Microsoft.Build.Tasks.Core.resources.dll" "en"
+    FileToCopy "$bootstrapBinDirectory\en\Microsoft.Build.Utilities.Core.resources.dll" "en"
+    FileToCopy "$bootstrapBinDirectory\en\MSBuild.resources.dll" "en"
```
Comment (rainersigwald) [inline scripts/Deploy-MSBuild.ps1:65]: Do we need en resources? I thought they were always embedded as default?

Comment (cdmihai) [reply scripts/Deploy-MSBuild.ps1:65]: That was my understanding as well until I got missing resource failures. I also deployed release bits, maybe those behave differently. I vaguely remember at some point, some email thread, where somebody suggested we don't treat en resources any differently to capture more possible bugs, so maybe I actually implemented that but forgot :)

File: scripts/Deploy-MSBuild.ps1 [hunk 2]
```diff
@@ -8,15 +8,30 @@ Param(
   [string] $runtime = "Desktop"
 )
 
+Set-StrictMode -Version "Latest"
+$ErrorActionPreference = "Stop"
```
Comment (BenVillalobos) [inline scripts/Deploy-MSBuild.ps1:12]: Wishlist item that may be too hard to implement: Building out the error functionality to delete the folder that was generated for this particular build, or essentially "reset" to the original binaries.

Comment (cdmihai) [reply scripts/Deploy-MSBuild.ps1:12]: I restore original binaries by adding the VS msbuild bin dir under git. If I were to implement restoration of original binaries, that's how I would do it. But I'd rather not do it in this PR :)

**Pull request merged**